include/solvers/multiphysics_interface.h

```diff
@@ -77,55 +75,149 @@ class MultiphysicsInterface
   {
     // Loop through all the elements in the physics map. Consequently, iphys is
     // an std::pair where iphys.first is the PhysicsID and iphys.second is the
     // AuxiliaryPhysics pointer. This is how the map can be traversed
     // sequentially.
     for (auto &iphys : physics)
       {
-        // Announce physic solved (verbosity = non_linear_solver.verbosity)
-        if (verbosity != Parameters::Verbosity::quiet)
-          announce_physics(iphys.first);
-
-        solve_physics(iphys.first, time_stepping_method);
+        // Ensure that iphys.first is present in solve_pre_fluid map
+        if (solve_pre_fluid.count(iphys.first) != 0)
+          {
+            if (solve_pre_fluid[iphys.first] == true)
+              {
+                // Announce physic solved (verbosity =
+                // non_linear_solver.verbosity)
+                if (verbosity != Parameters::Verbosity::quiet)
+                  announce_physics(iphys.first);
+
+                solve_physics(iphys.first, time_stepping_method);
+
+                // Embellish the console output
+                if (verbosity != Parameters::Verbosity::quiet)
+                  pcout << "-----------------------" << std::endl;
+              }
+          }
+        else
+          {
+            pcout
+              << "Resolution order is not specified for this auxiliary physic :"
+              << std::endl
+              << "will be solved after the fluid dynamics by default."
+              << std::endl
+              << "For clarity, please specify a resolution order in the map solve_pre_fluid defined in multiphysics_interface.h"
+              << std::endl
+              << "-------------" << std::endl;
+          }
       }
+
     for (auto &iphys : block_physics)
       {
-        // Announce physic solved (verbosity = non_linear_solver.verbosity)
-        if (verbosity != Parameters::Verbosity::quiet)
-          announce_physics(iphys.first);
-
-        solve_block_physics(iphys.first, time_stepping_method);
+        // Ensure that iphys.first is present in solve_pre_fluid map
+        if (solve_pre_fluid.count(iphys.first) != 0)
+          {
+            if (solve_pre_fluid[iphys.first] == true)
+              {
+                // Announce physic solved (verbosity =
+                // non_linear_solver.verbosity)
+                if (verbosity != Parameters::Verbosity::quiet)
+                  announce_physics(iphys.first);
+
+                solve_block_physics(iphys.first, time_stepping_method);
+
+                // Embellish the console output
+                if (verbosity != Parameters::Verbosity::quiet)
+                  pcout << "-----------------------" << std::endl;
+              }
+          }
+        else
+          {
+            pcout
+              << "Resolution order is not specified for this auxiliary physic :"
+              << std::endl
+              << "will be solved after the fluid dynamics by default."
```

Comment:
I don't understand this message. Can you please make it more clear?

Reply:
I think your suggestion is a good one. Anyway it's a bug that only developers should face :)!